synapse/lib/cell.py

```diff
@@ -3900,3 +3936,45 @@ async def getSystemInfo(self):
     async def getSpooledSet(self):
         async with await s_spooled.Set.anit(dirn=self.dirn, cell=self) as sset:
             yield sset
+
+    async def addSignalHandlers(self):
+        await s_base.Base.addSignalHandlers(self)
+
+        def sighup():
+            asyncio.create_task(self.reloadCell())
+
+        loop = asyncio.get_running_loop()
+        loop.add_signal_handler(signal.SIGHUP, sighup)
+
+    def addReloadFunc(self, name: str, func):
+        self._reloadfuncs[name] = func
+
+    def getReloadNames(self):
+        return tuple(self._reloadfuncs.keys())
+
+    async def reloadCell(self, name=None):
+        ret = {}
```

Comment:
This should probably return a dict of <name>: <retval> ( (ok, valu) ) tuples to account for exceptions etc.